src/main/java/org/tron/common/crypto/ECKey.java

```diff
@@ -1054,7 +1054,7 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || !(o instanceof ECKey)) {
+    if (this.getClass() != o.getClass()) {

```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "o" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=853&issues=AWOg828-mXhJxGlVU5lK&open=AWOg828-mXhJxGlVU5lK)

[ISSUE_KEY]: AWOg828-mXhJxGlVU5lK